### PR TITLE
Queuing and concurrent test run

### PR DESCRIPTION
**Pull request by chiayi**
Added queuing to test builds to prevent stockout. 
Test will run concurrently. Created separate cluster for jupyter+ray and rag. 

I switched the order of Jupyter and Ray test for race condition error. Run time of the test is now completely dependent on the longest running test, which currently is RAG: ~50mins including cluster create/delete time.

### CODE REVIEW

File: cloudbuild.yaml [hunk 1]
```diff
@@ -69,11 +142,12 @@ steps:
         echo "pass" > /workspace/gke_cluster_result.txt
     dir: 'infrastructure/'
     allowFailure: true
-    waitFor: ['validate platform', 'validate ray', 'validate jupyterhub', validate rag]
+    # waitFor: ['validate platform', 'validate ray', 'validate jupyterhub', validate rag]
```
Comment (yiyinglovecoding) [inline cloudbuild.yaml:145]: remove

Comment (chiayi) [reply cloudbuild.yaml]: done

File: cloudbuild.yaml [hunk 2]
```diff
@@ -282,28 +380,31 @@ steps:
...
-        -var=jupyter_service_account=jupyter-sa-$SHORT_SHA-$_BUILD_ID \
+        -var=jupyter_service_account=rag-jupyter-sa-$SHORT_SHA-$_BUILD_ID \
         -var=cloudsql_instance=pgvector-instance-$SHORT_SHA-$_BUILD_ID \
         -auto-approve -no-color
 
     allowFailure: true
     waitFor: ['test rag']
 
-  - id: 'cleanup gke cluster'
+  - id: 'cleanup jupyter jupyter ray gke cluster'
```
Comment (yiyinglovecoding) [inline cloudbuild.yaml:407]: I saw the cloudbuild failed at this step as well.
- Is it expected? looking at logs we didn't do anything here. 
```
Already have image (with digest): gcr.io/gke-ai-eco-dev/terraform
Switched to context "gke_gke-ai-eco-dev_us-east4_ml-c5d5cec-581-140c7add-cluster".
Switched to workspace "jupyterray".

No changes. No objects need to be destroyed.

Either you have not created any objects yet or the existing objects were
already deleted outside of Terraform.

Destroy complete! Resources: 0 destroyed.
Workspace "jupterray" doesn't exist.

You can create this workspace with the "new" subcommand 
``` 

Also the naming `jupyter jupyter`

**Comment by chiayi** [comment]
Discussed offline. We will be holding off on merging the queuing feature for now.

**Comment by spencer-p** [comment]
> We will be holding off on merging the queuing feature for now.

Let's close and re-open if needed.

**Pull request closed without merging**